pkg/chunk/util/util.go

```diff
@@ -15,7 +15,7 @@ import (
 // DoSingleQuery is the interface for indexes that don't support batching yet.
 type DoSingleQuery func(
 	ctx context.Context, query chunk.IndexQuery,
-	callback func(chunk.ReadBatch) bool,
+	callback func(chunk.IndexQuery, chunk.ReadBatch) bool,
```

Comment:
What do you think to replace `callback func(chunk.IndexQuery, chunk.ReadBatch) bool` with `callback Callback` given we have a type for that?

Reply:
Good spot: I missed that we already have that type.